fusor-ember-cli/app/controllers/openstack.js

```diff
@@ -1,8 +1,10 @@
 import Ember from 'ember';
 
 export default Ember.Controller.extend({
-  needs: ['undercloud-deploy'],
-  stepNumberOpenstack: Ember.computed.alias("controllers.undercloud-deploy.stepNumberOpenstack"),
+  needs: ['deployment', 'undercloud-deploy'],
+
+  stepNumberOpenstack: Ember.computed.alias("controllers.deployment.stepNumberOpenstack"),
   disableTabRegisterNodes: Ember.computed.alias("controllers.undercloud-deploy.disableTabRegisterNodes"),
-  disableTabAssignNodes: Ember.computed.alias("controllers.undercloud-deploy.disableTabAssignNodes")
+  disableTabAssignNodes: Ember.computed.alias("controllers.undercloud-deploy.disableTabAssignNodes"),
+  validOpenStack: true //TODO
```

Comment:
@tzumainn said on irc:
<tzumainn> so, I think you can use the function here: https://github.com/fusor/fusor/blob/master/fusor-ember-cli/app/controllers/assign-nodes.js#L8
<tzumainn> and the parameters you'd want are from: https://github.com/fusor/fusor/blob/master/fusor-ember-cli/app/models/deployment-role.js
<tzumainn> so given a role object, you can get its countParameterName and flavorParameterName and make sure the former > 0 and the latter is not empty

The two roles that are required in a minimum setup are 'compute' and 'controller'. So those two roles should have countParamName > 0 and flavorParamName != empty.

We should probably also check that there are enough machines (nodes) registered to fulfil the planed deployment. So maybe loop through the roles, add up the countParam that is associated with each flavor, and then make sure the number of nodes in that flavor is >= the counted planned roles?